src/model/CoilCoolingDXCurveFitOperatingMode.cpp

```diff
@@ -0,0 +1,545 @@
+    bool result = setDouble(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::RatioofInitialMoistureEvaporationRateandSteadyStateLatentCapacity, ratioofInitialMoistureEvaporationRateandSteadyStateLatentCapacity);
+    return result;
+  }
+
+  bool CoilCoolingDXCurveFitOperatingMode_Impl::setLatentCapacityTimeConstant(double latentCapacityTimeConstant) {
+    bool result = setDouble(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::LatentCapacityTimeConstant, latentCapacityTimeConstant);
+    return result;
+  }
+
+  bool CoilCoolingDXCurveFitOperatingMode_Impl::setNominalTimeforCondensateRemovaltoBegin(double nominalTimeforCondensateRemovaltoBegin) {
+    bool result = setDouble(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::NominalTimeforCondensateRemovaltoBegin, nominalTimeforCondensateRemovaltoBegin);
+    return result;
+  }
+
+  bool CoilCoolingDXCurveFitOperatingMode_Impl::setApplyLatentDegradationtoSpeedsGreaterthan1(bool applyLatentDegradationtoSpeedsGreaterthan1) {
+    if (applyLatentDegradationtoSpeedsGreaterthan1) {
+      return setString(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::ApplyLatentDegradationtoSpeedsGreaterthan1, "Yes");
+    } else {
+      return setString(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::ApplyLatentDegradationtoSpeedsGreaterthan1, "No");
+    }
+  }
+
+  bool CoilCoolingDXCurveFitOperatingMode_Impl::setCondenserType(const std::string& condenserType) {
+    bool result = setString(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::CondenserType, condenserType);
+    return result;
+  }
+
+  bool CoilCoolingDXCurveFitOperatingMode_Impl::setNominalEvaporativeCondenserPumpPower(double nominalEvaporativeCondenserPumpPower) {
+    bool result = setDouble(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::NominalEvaporativeCondenserPumpPower, nominalEvaporativeCondenserPumpPower);
+    return result;
+  }
+
+  void CoilCoolingDXCurveFitOperatingMode_Impl::autosizeNominalEvaporativeCondenserPumpPower() {
+    bool result = setString(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::NominalEvaporativeCondenserPumpPower, "autosize");
+    OS_ASSERT(result);
+  }
+
+  bool CoilCoolingDXCurveFitOperatingMode_Impl::setNominalSpeedNumber(int nominalSpeedNumber) {
+    bool result = setInt(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::NominalSpeedNumber, nominalSpeedNumber);
+    return result;
+  }
+
+  void CoilCoolingDXCurveFitOperatingMode_Impl::resetNominalSpeedNumber() {
+    bool result = setString(OS_Coil_Cooling_DX_CurveFit_OperatingModeFields::NominalSpeedNumber, "");
+    OS_ASSERT(result);
+  }
+
+  void CoilCoolingDXCurveFitOperatingMode_Impl::addSpeed(CoilCoolingDXCurveFitSpeed& speed) {
+    auto group = getObject<ModelObject>().pushExtensibleGroup().cast<WorkspaceExtensibleGroup>();
+    OS_ASSERT(!group.empty());
+    group.setPointer(OS_Coil_Cooling_DX_CurveFit_OperatingModeExtensibleFields::Speed,speed.handle());
+  }
```

Comment:
You are missing companion functions, like `removeSpeed`, perhaps SpeedIndex(CoilCoolingDXCurveFitSpeed&), setSpeeds(std::vector<CoilCoolingDXCurveFitSpeed>&) as well.

(It seems that CoilCoolingDXMutiSpeed has the same problem)